<a href="https://colab.research.google.com/github/anderson361163/AED_Impacta_Atv1/blob/main/atividade_agrupamentp_dispers%C3%A3o.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

*   Anderson Silva Machado
*   Fabiola Rosa Pedroso
*   Jhonatha Nunes Silveira

* Análise completa sobre o describe
* Fazer a amplitude interquartil do Valor de Venda e uma análise sobre o resultado apresentado
* Fazer o coeficiente de variação e uma análise sobre ele
* Fazer o agrupamento das top 10 bandeiras com maior valor de venda
* Fazer o agrupamento por estado e trazer os 5 estados com maior valor de venda e os 5 estados com menor valor de venda
* Trazer o agrupamento por mês com a média de valor de venda por mês

In [ ]:
# Import das bibliotecas
import pandas as pd
import numpy as np
import statistics

In [ ]:
base = pd.read_csv('/content/glp-2021-02.csv', delimiter=';', decimal=',')

# Análise describe

In [ ]:
base.describe().round(2)

Valor de Venda  Valor de Compra           Mes
count    89372.000000              0.0  89372.000000
mean        99.276313              NaN      9.583572
std          9.586874              NaN      1.702067
min         64.990000              NaN      7.000000
25%         93.000000              NaN      8.000000
50%         99.000000              NaN     10.000000
75%        105.000000              NaN     11.000000
max        140.000000              NaN     12.000000

Há um total de 89372 registros, com uma média de R\$ 99,27. O valor mais alto é 140 reais enquanto o mais novo é 64,99 reais. Verificamos que pelo menos 50% dos GLPs tem preço igual ou inferior a R\$ 99,00, com desvio padrão de R$ 9,58 para cima ou para baixo, mostrando uma váriação significativa do preço.

# Amplitude Interquartil

In [ ]:
desc = base.describe(percentiles=[0.15, 0.25, 0.50, 0.75]).round(2)

desc

Valor de Venda  Valor de Compra       Mes
count        89372.00              0.0  89372.00
mean            99.28              NaN      9.58
std              9.59              NaN      1.70
min             64.99              NaN      7.00
15%             89.99              NaN      8.00
25%             93.00              NaN      8.00
50%             99.00              NaN     10.00
75%            105.00              NaN     11.00
max            140.00              NaN     12.00

In [ ]:
q1 = desc['Valor de Venda'][5]
q3 = desc['Valor de Venda'][7]

amp_interquartil = q3 - q1
amp_interquartil

12.0

O centro da amostra apresenta uma oscilação da R$ 12,00.

# Coeficiente de variação

In [ ]:
media = base['Valor de Venda'].mean()
media

99.27631271541932

In [ ]:
desvio_padrao = desc['Valor de Venda'][2]
desvio_padrao

9.59

In [ ]:
coeficiente = round(desvio_padrao / media * 100, 1)
coeficiente

9.7

Os preços do GLP brasileiro apresenta uma variação de 9,7%

# Top 10 média valor de venda por bandeira

In [ ]:
base.groupby('Bandeira').mean()[['Valor de Venda']].sort_values(by=['Valor de Venda'], ascending=False).round(2)[:10]

Valor de Venda
Bandeira                            
FOGAS                         109.20
AMAZONGÁS                     107.18
SUPERGASBRAS                  107.17
COPAGAZ                       101.46
ULTRAGAZ                      100.10
LIQUIGÁS                      100.09
NACIONAL GÁS BUTANO            99.00
SUPERGASBRAS ENERGIA           98.48
BRANCA                         98.19
SERVGÁS                        96.69

A empresa com o maior valor médio do país foi a Fogas com RS 109,20, enquanto a décima empresa com maior valor médio é a Servgás com preço médio de R\$ 96,69. A diferênça é de R$ 12,51. A diferença entre o décimo é a bandeira com menor valor médio é apenas de alguns centavos.

# Top 5 estados com menor e maior valor médio de venda

In [ ]:
# Estados com maior valor
base.groupby('Estado - Sigla').mean()[['Valor de Venda']].sort_values(by=['Valor de Venda'], ascending=False).round(2)[:5]

Valor de Venda
Estado - Sigla                
MT                      118.69
AC                      113.44
RO                      113.26
AP                      111.88
RR                      110.66

Os 5 estados com maior valor médio são na maioria da região nordeste, com exceção de MT que tem o maior valor médio, com diferença de aproximadamente 8 reais de diferença com o quinto estado com maior valor que é Roraima

In [ ]:
# Estados com maior valor
base.groupby('Estado - Sigla').mean()[['Valor de Venda']].sort_values(by=['Valor de Venda'], ascending=True).round(2)[:5]

Valor de Venda
Estado - Sigla                
RJ                       89.30
PE                       91.23
BA                       92.19
SE                       92.26
DF                       94.30

O estado com menor valor é o RJ. Diferente dos 5 estados com maior valor não há uma predominância de uma região e a variação nos valores é menor

# Agrupamento por mês com média do valor

In [ ]:
lista = []
for linha in base['Data da Coleta']:
  lista.append(int(linha[3:5]))

In [ ]:
base['Mes'] = lista

In [ ]:
base.groupby('Mes').mean()[['Valor de Venda']].round(2)

Valor de Venda
Mes                
7             92.77
8             94.42
9             98.73
10           101.81
11           103.54
12           103.29

Em apenas um semestre a variação no valor de venda do gás, foi de pouco mais que 10 reais. A maior variação foi o valor do mês 8 e 9 que teve um aumento de R\$ 4,31 e a menor variação foi entre o mês 11 e 12 que diminuiu R$ 0,25